File: Java/6.29.java
/*
Write a program that simulates flipping a coin to make decisions. The input is how many decisions are needed, and the output is either heads or tails. Assume the input is a value greater than 0.

Ex: If the input is:

3
the output is:

tails
heads
tails
For reproducibility needed for auto-grading, seed the program with a value of 2. In a real program, you would seed with the current time. In that case, every program's output would be different, which is what is desired but can't be auto-graded.

Note: A common student mistake is to create an instance of Random before each call to rand.nextInt(). But seeding should only be done once, at the start of the program, after which rand.nextInt() can be called any number of times.

Your program must define and call the following method that randomly picks 0 or 1 and returns "heads" or "tails". Assume the value 0 represents "heads" and 1 represents "tails".
*/

import java.util.Scanner;
import java.util.Random;

public class LabProgram {
   
   /* Define your method here */
   public static String headsOrTails(Random rand) {
      int randomNum = rand.nextInt(2);
      
      if (randomNum == 0) {
         String ans = "heads";
         return ans;
      }
      else {
         String ans = "tails";
         return ans;
      }
   }
   
   public static void main(String[] args) {
      Scanner scan = new Scanner(System.in);
      Random rand = new Random(2); // Unique seed
      // Add more variables as needed

      /* Type your code here. */
      int decisions = scan.nextInt();      
      for (int i = 0; i < decisions; i++) {
         System.out.printf("%s\n", headsOrTails(rand));
      }      
   }
}

File: CPP/6.29.cpp
/*
Write a program whose inputs are three integers, and whose outputs are the largest of the three values and the smallest of the three values.

Ex: If the input is:

7 15 3
the output is:

largest: 15 
smallest: 3
Your program must define and call the following two functions. The LargestNumber function should return the largest number of the three input values. 
The SmallestNumber function should return the smallest number of the three input values.
*/

#include <iostream>
using namespace std;

/* Define your functions here */ 
int LargestNumber(int num1, int num2, int num3) {
   return max(num1, max(num2, num3));
}

int SmallestNumber(int num1, int num2, int num3) {
   return min(num1, min(num2, num3));
}

int main() {
   /* Type your code here. Your code must call the functions.  */
   int userInput1, userInput2, userInput3;
   
   cin >> userInput1 >> userInput2 >> userInput3;
   
   printf("largest: %d\n", LargestNumber(userInput1, userInput2, userInput3));
   printf("smallest: %d\n", SmallestNumber(userInput1, userInput2, userInput3));

   return 0;
}
